cpp/src/arrow/compute/kernels/scalar_validity_test.cc

```diff
@@ -0,0 +1,151 @@
+  UnaryFunction func_;
+};
+
+TEST_F(TestValidityKernels, ArrayIsValid) {
+  func_ = arrow::compute::IsValid;
+
+  this->AssertUnary("[]", "[]");
+  this->AssertUnary("[null]", "[false]");
+  this->AssertUnary("[1]", "[true]");
+  this->AssertUnary("[null, 1, 0, null]", "[false, true, true, false]");
+}
+
+TEST_F(TestValidityKernels, ArrayIsValidBufferPassthruOptimization) {
+  Datum arg = ArrayFromJSON(boolean(), "[null, 1, 0, null]");
+  ASSERT_OK_AND_ASSIGN(auto validity, arrow::compute::IsValid(arg));
+  ASSERT_EQ(validity.array()->buffers[1], arg.array()->buffers[0]);
+}
+
+TEST_F(TestValidityKernels, ScalarIsValid) {
+  func_ = arrow::compute::IsValid;
+
+  AssertUnary(Datum(19.7), Datum(true));
+  AssertUnary(MakeNullScalar(float64()), Datum(false));
+}
+
+TEST_F(TestValidityKernels, ArrayIsNull) {
+  func_ = arrow::compute::IsNull;
+
+  this->AssertUnary("[]", "[]");
+  this->AssertUnary("[null]", "[true]");
+  this->AssertUnary("[1]", "[false]");
+  this->AssertUnary("[null, 1, 0, null]", "[true, false, false, true]");
+}
+
+TEST_F(TestValidityKernels, DISABLED_ScalarIsNull) {
+  func_ = arrow::compute::IsNull;
+
+  AssertUnary(Datum(19.7), Datum(false));
+  AssertUnary(MakeNullScalar(float64()), Datum(true));
+}
+
+class IsValidProperty : public ScalarFunctionPropertyMixin {
+ public:
+  std::shared_ptr<ScalarFunction> GetFunction() override {
+    return internal::checked_pointer_cast<ScalarFunction>(
+        *GetFunctionRegistry()->GetFunction("is_valid"));
+  }
+
+  Result<std::shared_ptr<Scalar>> Contract(const ScalarVector& args,
+                                           const FunctionOptions*) override {
+    return std::make_shared<BooleanScalar>(args[0]->is_valid);
+  }
+};
+
+TEST_P(IsValidProperty, TestIsValidProperty) { Validate(); }
+
+INSTANTIATE_TEST_SUITE_P(IsValidPropertyTest, IsValidProperty,
+                         ScalarFunctionPropertyTestParam::Values({
+                             {0, 0.0},
+                             {1, 0.0},
+                             {2, 0.0},
+                             {1024, 0.25},
+                         }));
+
+class IsNullProperty : public ScalarFunctionPropertyMixin {
+ public:
+  std::shared_ptr<ScalarFunction> GetFunction() override {
+    return internal::checked_pointer_cast<ScalarFunction>(
+        *GetFunctionRegistry()->GetFunction("is_null"));
+  }
+
+  Result<std::shared_ptr<Scalar>> Contract(const ScalarVector& args,
+                                           const FunctionOptions*) override {
+    return std::make_shared<BooleanScalar>(!args[0]->is_valid);
+  }
+};
+
+TEST_P(IsNullProperty, TestIsNullProperty) { Validate(); }
+
+INSTANTIATE_TEST_SUITE_P(IsNullPropertyTest, IsNullProperty,
+                         ScalarFunctionPropertyTestParam::Values({
+                             {0, 0.0},
+                             {1, 0.0},
+                             {2, 0.0},
+                             {1024, 0.25},
```

Comment:
Am I wrong, or is this generating a new test subclass for simply different parameter values? This does not seem sound, especially as it will blow up compile times.

Reply:
It isn't generating new classes. `TYPED_TEST_SUITE` does this (one template instantiation per type) but `INSTANTIATE_TEST_SUITE_P` only iterates over the provided values (which in this case are all `ScalarFunctionPropertyTestParam`)